pkg/helm/helm.go

```diff
@@ -185,17 +185,14 @@ func (r *Impl) RegistryLogin(hostname string, username string, password string,
 }
 
 // RegistryLogout to docker registry v2
-func (r *Impl) RegistryLogout(hostname string, username string) error {
-	if username != "" {
-		cfg, err := getConfig("")
-		if err != nil {
-			return err
-		}
-		logout := action.NewRegistryLogout(cfg)
-		var buf bytes.Buffer
-		return logout.Run(&buf, hostname)
+func (r *Impl) RegistryLogout(hostname string) error {
+	cfg, err := getConfig("")
```

Comment:
If the method gets called but DOCKER_USERNAME and DOCKER_PASSWORD are not specified (e.g. when using local registry that does not need any credentials) does this function fail in `getConfig("")` ? Or can the `logoug.Run` deal with this situation?

Reply:
It isn't directly related to not having username/password. It throws an error "not logged in"  if logout is called but no login was done before.  I think is the expected behavior. 

Caller code should only call logout if login returned successfully. I think that changing RegistryLogin to not silently ignore being called without username/password would be the right approach. What you agree?

Reply:
I updated the code according to the logic described in the message above